File: wayne-agent/apps/shared/src/product-copy.ts
/**
 * Legacy runtime names (Wayne / Hermes) must not appear in user-visible copy.
 * Use at UI boundaries when surfacing engine, API, or plugin text.
 */

const BRAND_REPLACEMENTS: readonly [RegExp, string][] = [
  [/\bMotor Wayne\b/gi, 'Work4You'],
  [/\bWayne Agent\b/gi, 'Work4You'],
  [/\bHermes Agent\b/gi, 'Work4You'],
  [/\bWayne Console\b/gi, 'Work4You Console'],
  [/\bWayne Dashboard\b/gi, 'Work4You'],
  [/\bNous Research\b/gi, 'Work4You'],
  [/\bNous Portal\b/gi, 'Work4You account'],
  [/\bNous subscription\b/gi, 'Work4You subscription'],
  [/\bNous credits\b/gi, 'Work4You credits'],
  [/\bWayne\b/gi, 'Work4You'],
  [/\bHermes\b/gi, 'Work4You']
]

/** Strip legacy product/runtime brands from strings shown to the user. */
export function sanitizeProductCopy(text: string): string {
  if (!text) {
    return text
  }

  let out = text
  for (const [pattern, replacement] of BRAND_REPLACEMENTS) {
    out = out.replace(pattern, replacement)
  }

  return out
}

File: wayne-agent/work4you_cli/product_copy.py
"""Sanitize user-visible strings — strip legacy Wayne/Hermes/Nous product names."""

from __future__ import annotations

import re

from work4you_cli.relay_free_model import W4Y_DOCS_BASE, W4Y_LOGIN_URL

# Order matters: longer / specific phrases before bare tokens.
_BRAND_REPLACEMENTS: tuple[tuple[re.Pattern[str], str], ...] = (
    (re.compile(r"\bMotor Wayne\b", re.I), "Work4You"),
    (re.compile(r"\bWayne Agent\b", re.I), "Work4You"),
    (re.compile(r"\bHermes Agent\b", re.I), "Work4You"),
    (re.compile(r"\bNous Research\b", re.I), "Work4You"),
    (re.compile(r"\bNous Portal\b", re.I), "Work4You account"),
    (re.compile(r"\bNous subscription\b", re.I), "Work4You subscription"),
    (re.compile(r"\bNous credits\b", re.I), "Work4You credits"),
    (re.compile(r"\bWayne Setup\b", re.I), "Work4You Setup"),
    (re.compile(r"\bWayne\b", re.I), "Work4You"),
    (re.compile(r"\bHermes\b", re.I), "Work4You"),
)

_LEGACY_DOCS_PREFIX = "https://hermes-agent.nousresearch.com/docs"


def product_docs_url(path: str = "") -> str:
    """Map a legacy Hermes docs path to Work4You documentation."""
    segment = (path or "").strip().lstrip("/")
    return f"{W4Y_DOCS_BASE}/{segment}" if segment else W4Y_DOCS_BASE


def sanitize_product_copy(text: str) -> str:
    if not text:
        return text
    out = text.replace(_LEGACY_DOCS_PREFIX, W4Y_DOCS_BASE)
    for pattern, replacement in _BRAND_REPLACEMENTS:
        out = pattern.sub(replacement, out)
    return out


__all__ = ["W4Y_DOCS_BASE", "W4Y_LOGIN_URL", "product_docs_url", "sanitize_product_copy"]
